src/Cli/dotnet/commands/dotnet-new/PostActions/DotnetSlnPostActionProcessor.cs

```diff
@@ -27,8 +27,12 @@ public DotnetSlnPostActionProcessor(Func<string, IReadOnlyList<string>, string?,
 
         internal static IReadOnlyList<string> FindSolutionFilesAtOrAbovePath(IPhysicalFileSystem fileSystem, string outputBasePath)
         {
-            return FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.sln")
-                ?? FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.slnx");
+            var slnFiles = FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.sln");
+            if (slnFiles.Count > 0)
+            {
+                return slnFiles;
+            }
+            return FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.slnx");
```

Comment:
That if statement in the middle looks a bit weird to me. What about instead doing something like:

```suggestion
            return [.. FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.sln"), ..FileFindHelpers.FindFilesAtOrAbovePath(fileSystem, outputBasePath, "*.slnx")]
```
?